src/main/java/org/jenkinsci/plugins/p4/PerforceScm.java

```diff
@@ -220,8 +224,22 @@ public String getKey() {
 			logger.fine("No credential for perforce");
 			return null;
 		}
+
+		// Retrieve item from request
+		StaplerRequest req = Stapler.getCurrentRequest();
+		Job job = req == null ? null : req.findAncestorObject(Job.class);
```

Comment:
@jglick Thanks for the feedback. I cannot find another way to get the context (_Item_ or _ItemGroup_ ) from this method. What do you suggest ?